### PR TITLE
Added Overview of Request and Response Headers, expanded spec and examples section

### PR DESCRIPTION
**Pull request by acquiachrisnagy**
- Spec section follows top-down approach of defining components 
- Example section follows bottom-up approach of building up the header values

### CODE REVIEW

**Comment by kevinhankens** [comment]
Thank you for this! I wish I had this a week ago :D

I'm going to take a closer look and I'll add comments inline if I see anything.

File: README.md [hunk 1]
```diff
@@ -45,45 +45,76 @@ TBD, see the Hawk spec for inspiration
 
 Production services implementing this spec must only accept requests using HTTPS.
 
-## Overview of Header and Signature
+## Overview of Request and Response Headers
```
Comment (kevinhankens) [inline README.md:54]: Might want to update the table of contents links above.

File: README.md [hunk 2]
```diff
@@ -45,45 +45,76 @@ TBD, see the Hawk spec for inspiration
...
-#### Secret Key
+## Overview of Request Header and Signature
 
-The secret key should be a 256 to 512 bit binary value. The secret key will normally be stored as a base64-encoded or hex-encoded string representation, but must be decoded to the binary value before use.
+The pseudocode below illustrates construction of the HTTP "Authorization" header and signature:
 
-#### Authorization Header
+```
+HMACAuthorization = "acquia-http-hmac" + " " +
+                "realm=" + Provider + "," +
```
Comment (kevinhankens) [inline README.md]: It might be nice to mention in the pseudo code that each attribute value is urlencoded. Up to you obviously.

**Comment by acquiachrisnagy** [comment]
I've updated to table of contents and some functions to the pseudo code to illustrate how the attributes are encoded. Let me know if you have any other suggestions

**Comment by acquiachrisnagy** [comment]
Thanks for the suggestions

**Comment by kevinhankens** [comment]
Sweet. +1 from me. @nickveenhof @cpliakas ok to merge?

**Comment by cpliakas** [comment]
Doc improvements are best made by people who wrestle with implementations for the first time, so I say go for it!

**Comment by kevinhankens** [comment]
Done. Thanks!!

**Pull request merged**